### PR TITLE
fix(select): clear select if no option matches value

### LINKED ISSUE #2109: Select should clear selection if setting value programmatically to nonexistent option value
Right now if you set the select value to an option value that doesn't exist, nothing happens. To emulate native selects, it should clear the select.

### PR DESCRIPTION
**Pull request by kara**
Closes #2109 

r: @jelbourn

### CODE REVIEW

**Comment by kara** [comment]
Noticed a weird animation thing, want to look into it

File: src/lib/select/select.ts [hunk 1]
```diff
@@ -378,6 +374,28 @@ export class MdSelect implements AfterContentInit, ControlValueAccessor, OnDestr
     scrollContainer.scrollTop = this._scrollTop;
   }
 
+  /**
+   * Sets the selected option based on a value. If no option can be
+   * found with the designated value, the select trigger is cleared.
+   */
+  private _setSelectionByValue(value: any): void {
+    this.options.forEach((option: MdOption) => {
```
Comment (jelbourn) [inline src/lib/select/select.ts]: I don't think you need the type here, just
```ts
this.options.forEach(option => {
```
Should be equivalent

File: src/lib/select/select.ts [hunk 2]
```diff
@@ -378,6 +374,28 @@ export class MdSelect implements AfterContentInit, ControlValueAccessor, OnDestr
...
+      if (option.value === value) {
+        option.select();
```
Comment (jelbourn) [inline src/lib/select/select.ts]: Stop iterating if you find a value?

**Comment by kara** [comment]
@jelbourn Comments addressed!

File: src/lib/select/select.ts [hunk 3]
```diff
@@ -378,6 +374,30 @@ export class MdSelect implements AfterContentInit, ControlValueAccessor, OnDestr
     scrollContainer.scrollTop = this._scrollTop;
   }
 
+  /**
+   * Sets the selected option based on a value. If no option can be
+   * found with the designated value, the select trigger is cleared.
+   */
+  private _setSelectionByValue(value: any): void {
+    const optionArr = this.options.toArray();
```
Comment (jelbourn) [inline src/lib/select/select.ts]: Just `options`- I have a strong aversion to suffixing with types.

**Comment by jelbourn** [comment]
LGTM otherwise

File: src/lib/select/select.ts (hunk 3 above)
Comment (kara) [inline src/lib/select/select.ts]: Sure, changing!

**Pull request merged**